### PR TITLE
Move rego files into their respective apps

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

This is the promised sequel to #7734. After this change, the `iam` app will no longer contain any code specific to other apps.

To make this work, the `/api/auth/rules` endpoint will now construct the OPA bundle from a set of paths, which will be populated by `load_app_permissions`.

Move OPA test files accordingly. Fortunately, `opa test` accepts multiple directories, so it is trivial to adapt the testing instructions.

Make the necessary adaptations to `generate_tests.py` to search for test generators in every app. The original parameters of `generate_tests.py` don't really make sense when there are multiple `rules` directory, so remove them.

Instead, add a new `--apps-dir` parameter. This parameter isn't really needed to test the open source version of CVAT, but I expect it to be useful for testing the Enterprise version.

In addition, add some safety checks to `generate_tests.py`:

* Make sure that we find at least one test generator.

* Propagate exceptions from `call_generator` into the main thread.

### How has this been tested?

I tested the updated commands from the documentation manually, and examined the rules bundle returned by `/api/auth/rules` to ensure that it still contains all the `.rego` files.

### Checklist

- [x] I submit my changes into the `develop` branch
- ~~[ ] I have created a changelog fragment~~ 
- [x] I have updated the documentation accordingly
- ~~[ ] I have added tests to cover my changes~~
- [x] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- ~~[ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
  - Enhanced permission management by dynamically loading OPA rules from specified paths.
- **Bug Fixes**
  - Improved file path handling in documentation for storing `.rego` files.
- **Documentation**
  - Updated test generation and running instructions to be more efficient and consistent.
  - Refined documentation on IAM user roles to reflect new file paths.
- **Chores**
  - Updated GitHub workflows to optimize test generation and linting processes.
- **Refactor**
  - Streamlined handling of OPA rules paths in the codebase.
- **Style**
  - Added `.gitignore` entry to exclude files generated by test scripts.

### CODE REVIEW

**Comment by nmanovic** [comment]
@coderabbitai review

**Comment by nmanovic** [comment]
@coderabbitai review

File: .github/workflows/regallint.yml [hunk 1]
```diff
@@ -9,4 +9,4 @@ jobs:
         uses: StyraInc/setup-regal@v0.2.0
         with:
           version: v0.11.0
-      - run: regal lint --format=github cvat/apps/iam/rules
+      - run: regal lint --format=github cvat/apps/*/rules
```
Comment (nmanovic) [inline .github/workflows/regallint.yml:12]: Do you think we can update regal in the PR? https://github.com/StyraInc/regal/releases/tag/v0.21.3

**Review by nmanovic — APPROVED** [review]
LGTM

File: .github/workflows/regallint.yml (hunk 1 above)
Comment (SpecLad) [reply .github/workflows/regallint.yml:12]: I ended up doing that in a separate PR: #7824.

**Pull request merged**